lit_nlp/examples/models/tydi.py

```diff
@@ -0,0 +1,176 @@
+"""LIT wrappers for T5, supporting both HuggingFace and SavedModel formats."""
+import re
+from typing import List
+
+import attr
+from lit_nlp.api import model as lit_model
+from lit_nlp.api import types as lit_types
+from lit_nlp.examples.models import model_utils
+from lit_nlp.lib import utils
+
+import tensorflow as tf
+# tensorflow_text is required for T5 SavedModel
+import tensorflow_text  # pylint: disable=unused-import
+import transformers
+from transformers import BertTokenizer
+from transformers import FlaxBertForQuestionAnswering
+
+from rouge_score import rouge_scorer
+
+JsonDict = lit_types.JsonDict
+
+
+def masked_token_mean(vectors, masks):
+  """Mean over tokens.
+
+  Args:
+    vectors: <tf.float32>[batch_size, num_tokens, emb_dim]
+    masks: <tf.int32>[batch_size, num_tokens]
+
+  Returns:
+    <tf.float32>[batch_size, emb_dim]
+  """
+  masks = tf.cast(masks, tf.float32)
+  weights = masks / tf.reduce_sum(masks, axis=1, keepdims=True)
+  return tf.reduce_sum(vectors * tf.expand_dims(weights, axis=-1), axis=1)
+
+
+@attr.s(auto_attribs=True, kw_only=True)
+class T5ModelConfig(object):
+  """Config options for a T5 generation model."""
+  # Input options
+  inference_batch_size: int = 4
+  # Generation options
+  beam_size: int = 4
+  max_gen_length: int = 50
+  num_to_generate: int = 1
+  # Decoding options
+  token_top_k: int = 10
+  output_attention: bool = False
+
+
+def validate_t5_model(model: lit_model.Model) -> lit_model.Model:
+  """Validate that a given model looks like a T5 model.
+
+  This checks the model spec at runtime; it is intended to be used before server
+  start, such as in the __init__() method of a wrapper class.
+
+  Args:
+    model: a LIT model
+
+  Returns:
+    model: the same model
+
+  Raises:
+    AssertionError: if the model's spec does not match that expected for a T5
+    model.
+  """
+  # Check inputs
+  ispec = model.input_spec()
+  assert "input_text" in ispec
+  assert isinstance(ispec["input_text"], lit_types.TextSegment)
+  if "target_text" in ispec:
+    assert isinstance(ispec["target_text"], lit_types.TextSegment)
+
+  # Check outputs
+  ospec = model.output_spec()
+  assert "output_text" in ospec
+  assert isinstance(
+      ospec["output_text"],
+      (lit_types.GeneratedText, lit_types.GeneratedTextCandidates))
+  assert ospec["output_text"].parent == "target_text"
+
+  return model
```

Comment:
Import these from the T5 module using the pattern mentioned above for getting around [Python Style Guide's import restrictions](https://google.github.io/styleguide/pyguide.html#22-imports).

```python
import ./t5 as t5

T5ModelConfig = t5.T5ModelConfig

masked_token_mean = t5.masked_token_mean
validate_t5_model = t5.validate_t5_model
```

Also, are we sure we even need these since this is a Flax model?